docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,214 @@
+    * **thin-edge** puts the measurement into the context of the corresponding metric and sends it to the cloud
+  * **Command**:
+    * a **command** is represented with one or more values received from the **cloud**
+    * **thin-edge** provides those values to the **domain application** and **protocol drivers** as **data point** values
+  * **Events**:
+    * for events **thin-edge** does not get in touch with **data points** at all;<br/>
+      instead any event related **data point** and it's value is evaluated by the event producer internally (i.E. the **domain application** or **protocol drivers**)
+    * the event producer sends a coresponding notification to **thin-edge** whenever an event shall be raised
+  * **Alarms**:
+    * similar to **events**; but in addition, the alarm producer can send a notification to **thin-edge** to clear an **alarm** 
+    * **thin-edge** raises or clears the alarm in the cloud
+
+## thin-edge device management concept
+
+**thin-edge** maps cloud's different **Device Management** functionalities to different resources of the device:
+  * **Software Management**:
+    * software packages are installable units on the device
+    * those units could be
+      * the **domain application**
+      * parts from **OS / Libs / Runtime**, or the whole thing as one
+    * examples for software packages are
+      * packages for a Linux Packages Managers (e.g. for Debian, ...)
+      * container images (e.g. for Docker)
+      * simple ZIP files
+      * custom specific files/packages
+  * **Configuration Management**:
+    * a configuration is a text file or a binary file
+    * those configurations could be
+      * configuration file(s) of the **domain application**
+      * one or more configurations file of **OS / Libs / Runtime**
+  * **Log Management**:
+    * a Log is a log file, could be
+      * log file(s) of the **domain application**
+      * one or more log files of **OS / Libs / Runtime**
+
+**thin-edge** realizes cloud's **Device Management** based on **plugins**.
+  * a **plugin** encapsulates and manages access to _ressources_ and _services_ of the device, as e.g.
+      * _software management_ accesses the device's _package manager_
+      * _configuration management_ accesses device's _configuration files_
+  * a **plugin** can be
+      * an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+      * or a thin-edge built-in software component (e.g. as for _software management_)
+  * usually a **plugin** runs on the **main-device**; thus it can access the _resources_ of the **main-device** directly
+  * to access _ressources_ of an **external child-device** a **plugin** needs another component, referred as **child-device agent**
+
+The figure below illustrates the concept of **plugins** and **child-devices agents**.
+
+![Plugin Concept](images/plugin-concept.svg)
+
+### Child-Device Agent
```

Comment:
I have trouble understanding this.
What is the problem we solve? What is the main solution?
(The text seems to jump into some more "esoteric" usage patterns right in the beginning. Some of these I would disagree)

Reply:
With the latest version the section 'Plugins' and subsection 'Child-Device Agent' are restructured and precised.

See https://github.com/thin-edge/thin-edge.io/blob/e4c57206351183fc906717cae3ba2a3fea0dcb93/docs/src/architecture/domain-model.md#plugins

@vaillant , does that solve your confusion?